.github/workflows/preview-deploy.yml

```diff
@@ -1,12 +1,14 @@
 name: Preview
 
-# run when a pull request is made
-on: [pull_request]
+on:
+  pull_request:
 
 jobs:
   code-engine:
-    # Prevents action to be executed by forks
-    if: github.event.pull_request.head.repo.full_name == github.repository
+    # Run the job if it's not triggered by a PR or if the PR is from the same repository
+    # This is because this job requires a secret, and secrets cannot be shared with forks
+    if: github.event.pull_request == null || github.event.pull_request.head.repo.full_name == github.repository
```

Comment:
same question as above, if this is set to only run`on: pull_request` when would `github.event.pull_request == null`?